be/src/io/fs/local_file_reader.cpp

```diff
@@ -39,16 +34,18 @@ LocalFileReader::~LocalFileReader() {
 }
 
 Status LocalFileReader::close() {
-    bool expected = false;
-    if (_closed.compare_exchange_strong(expected, true)) {
-        _file_handle.reset();
-        DorisMetrics::instance()->local_file_open_reading->increment(-1);
+    if (!closed()) {
```

Comment:
Close method maybe called concurrently, the _fd maybe set to -1 in another thread. I think we'd better use lock here.